### ISSUE TITLE
Player crouch should work better with colliders

### ISSUE THREAD

**Issue by Malcolmnixon** [GodotVR/godot-xr-tools#384]
Currently when the player starts or stops crouching, the player body instantly snaps to the new height. Shrinking isn't usually a problem, but growing the player body instantly to be over 2x the height can actually push more than half the players body beneath the floor causing the player to fall through the floor.

Additionally the player body does not check whether there is room to grow, so the player is able to stand up inside a cave. Instead the body should check (probably by move_and_collide test) to see how far the player can grow.

**Comment by DigitalN8m4r3** [comment]
indeed, now it makes sense why i ended up havin always issues in my hidemechanics, as they are built on the idea of scalin the player to crouch and back to standing. Good catch
